File: 백준/Silver/1920. 수 찾기/수 찾기.py
n = int(input())
arr = list(map(int, input().split()))
m = int(input())
line = list(map(int, input().split()))

def binary_search(arr, target):
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == target:
            return True
        elif arr[mid] > target:
            right = mid - 1
        else:
            left = mid + 1
    return False

arr.sort()

for i in line:
    if binary_search(arr, i):
        print(1)
    else:
        print(0)

File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.*;
import java.util.Arrays;
import java.util.Collections;
import java.util.StringTokenizer;

public class Main {
    static int N,M;
    static Integer[] arr;
    static int start,end,middle;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        arr = new Integer[N];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(arr);

        M = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < M; i++) {
            int a = Integer.parseInt(st.nextToken());
            System.out.println(binarySearch(a));
        }

    }

    static int binarySearch(int target) {
        start = 0;
        end = N - 1;

        while (start <= end) {
            middle = (start + end) / 2;

            if (arr[middle] < target) {
                start = middle + 1;
            } else if(arr[middle] > target) {
                end = middle - 1;
            } else {
                return 1;
            }
        }
        return 0;
    }
}